app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,165 @@
+
+  private var itemSelectedPosition = -1
+  private var selectedAnswerIndex = -1
+  private var selectedHtmlStringList = mutableListOf<String>()
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+    return when (viewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<MultipleChoiceInteractionItemsBinding>(
+            inflater,
+            R.layout.multiple_choice_interaction_items,
+            parent,
+            /* attachToParent= */ false
+          )
+        MultipleChoiceViewHolder(binding)
+      }
+      VIEW_TYPE_ITEM_SELECTION -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<ItemSelectionInteractionItemsBinding>(
+            inflater,
+            R.layout.item_selection_interaction_items,
+            parent,
+            /* attachToParent= */ false
+          )
+        ItemSelectionViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> (holder as MultipleChoiceViewHolder).bind(
+        itemList[position].htmlContent,
+        position,
+        itemSelectedPosition
+      )
+      VIEW_TYPE_ITEM_SELECTION -> (holder as ItemSelectionViewHolder).bind(
+        itemList[position]
+      )
+    }
+  }
+
+  // Determines the appropriate ViewType according to the interaction type.
+  override fun getItemViewType(position: Int): Int {
+    return if (customizationArgs.interactionId == "ItemSelectionInput") {
+      if (customizationArgs.maxAllowableSelectionCount > 1) {
+        VIEW_TYPE_ITEM_SELECTION
+      } else {
+        VIEW_TYPE_MULTIPLE_CHOICE
+      }
+    } else {
+      VIEW_TYPE_MULTIPLE_CHOICE
+    }
+  }
+
+  override fun getItemCount(): Int {
+    return itemList.size
+  }
+
+  private inner class ItemSelectionViewHolder(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(selectionContentViewModel: SelectionContentViewModel) {
+      binding.setVariable(BR.htmlContent, selectionContentViewModel.htmlContent)
+      binding.executePendingBindings()
+      val htmlResult: Spannable = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
+        selectionContentViewModel.htmlContent,
+        binding.root.item_selection_contents_text_view
+      )
+      binding.root.item_selection_contents_text_view.text = htmlResult
+      binding.root.item_selection_checkbox.isChecked = selectionContentViewModel.isAnswerSelected
+      Log.d(
+        INTERACTION_ADAPTER_TAG,
+        "You cannot select more than " + selectedHtmlStringList.size + " "+binding.root.item_selection_checkbox
```

Comment:
Spacing issues:

`"You cannot select more than " + selectedHtmlStringList.size + " " + binding.root.item_selection_checkbox`

Reply:
This log was for debug. its removed.